Source/SwiftLintFramework/Rules/Style/ExplicitSelfRule.swift

```diff
@@ -0,0 +1,199 @@
+import Foundation
+import SourceKittenFramework
+
+public struct ExplicitSelfRule: CorrectableRule, ConfigurationProviderRule, CompilerArgumentRule, OptInRule,
+    AutomaticTestableRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "explicit_self",
+        name: "Explicit Self",
+        description: "Instance variables and functions should be explicitly accessed with 'self.'.",
+        kind: .style,
+        nonTriggeringExamples: [
+            """
+            struct A {
+                func f1() {}
+                func f2() {
+                    self.f1()
+                }
+            }
+            """,
+            """
+            struct A {
+                let p1: Int
+                func f1() {
+                    _ = self.p1
+                }
+            }
+            """
+        ],
+        triggeringExamples: [
+            """
+            struct A {
+                func f1() {}
+                func f2() {
+                    ↓f1()
+                }
+            }
+            """,
+            """
+            struct A {
+                let p1: Int
+                func f1() {
+                    _ = ↓p1
+                }
+            }
+            """
+        ],
+        corrections: [
+            """
+            struct A {
+                func f1() {}
+                func f2() {
+                    ↓f1()
+                }
+            }
+            """:
+            """
+            struct A {
+                func f1() {}
+                func f2() {
+                    self.f1()
+                }
+            }
+            """,
+            """
+            struct A {
+                let p1: Int
+                func f1() {
+                    _ = ↓p1
+                }
+            }
+            """:
+            """
+            struct A {
+                let p1: Int
+                func f1() {
+                    _ = self.p1
+                }
+            }
+            """
+        ],
+        requiresFileOnDisk: true
```

Comment:
This property isn't very clear to me, which files does it require?

Reply:
Requires that `File`s that are passed in have a valid `path` on disk, unlike other uses of File, some of which are initialized with `File(contents:)` and have no path on disk.

Reply:
Definitely a large change.